From aa8cf9f5441eb29dabc174075d4efde68a940953 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Fri, 13 May 2022 08:26:14 +0300
Subject: [PATCH] imx8-isi: Fix the 5.15.5 stream capture color distortion

cap:
Restore the order format of the mxc_isi_src_formats[] array.
Move a new format to the env of array.
5.15.5 regression

hw:
Restore the available list format is_yuv().
5.15.5 regression

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 drivers/staging/media/imx/imx8-isi-cap.c | 16 ++++++++--------
 drivers/staging/media/imx/imx8-isi-hw.c  |  1 +
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/media/imx/imx8-isi-cap.c b/drivers/staging/media/imx/imx8-isi-cap.c
index 4457d5de2896..d902c32d8a57 100644
--- a/drivers/staging/media/imx/imx8-isi-cap.c
+++ b/drivers/staging/media/imx/imx8-isi-cap.c
@@ -41,14 +41,6 @@ static int mxc_isi_cap_streamoff(struct file *file, void *priv,
  */
 struct mxc_isi_fmt mxc_isi_src_formats[] = {
 	{
-		.name		= "UYVY-16",
-		.fourcc		= V4L2_PIX_FMT_UYVY,
-		.depth		= { 16 },
-		.color		= MXC_ISI_OUT_FMT_YUV422_1P8P,
-		.memplanes	= 1,
-		.colplanes	= 1,
-		.mbus_code	= MEDIA_BUS_FMT_UYVY8_2X8,
-	}, {
 		.name		= "RGB32",
 		.fourcc		= V4L2_PIX_FMT_RGB32,
 		.depth		= { 32 },
@@ -60,6 +52,14 @@ struct mxc_isi_fmt mxc_isi_src_formats[] = {
 		.depth		= { 32 },
 		.memplanes	= 1,
 		.colplanes	= 1,
+	}, {
+		.name		= "UYVY-16",
+		.fourcc		= V4L2_PIX_FMT_UYVY,
+		.depth		= { 16 },
+		.color		= MXC_ISI_OUT_FMT_YUV422_1P8P,
+		.memplanes	= 1,
+		.colplanes	= 1,
+		.mbus_code	= MEDIA_BUS_FMT_UYVY8_2X8,
 	}
 };
 
diff --git a/drivers/staging/media/imx/imx8-isi-hw.c b/drivers/staging/media/imx/imx8-isi-hw.c
index 08b57fe28f86..5796deda3df8 100644
--- a/drivers/staging/media/imx/imx8-isi-hw.c
+++ b/drivers/staging/media/imx/imx8-isi-hw.c
@@ -125,6 +125,7 @@ static bool is_yuv(u32 pix_fmt)
 	if ((pix_fmt == V4L2_PIX_FMT_YUYV)  ||
 	    (pix_fmt == V4L2_PIX_FMT_YUV32) ||
 	    (pix_fmt == V4L2_PIX_FMT_YUV444M) ||
+	    (pix_fmt == V4L2_PIX_FMT_YUV24) ||
 	    (pix_fmt == V4L2_PIX_FMT_NV12))
 		return true;
 	else
-- 
2.17.1

